plugins/catalyst/lib/Conversion/MQTOptToCatalystQuantum/MQTOptToCatalystQuantum.cpp

```diff
@@ -0,0 +1,1198 @@
+
+    // Output type setup
+    const Type qubitType =
+        catalyst::quantum::QubitType::get(rewriter.getContext());
+    const SmallVector<Type> qubitTypes(
+        inQubitsValues.size() + inCtrlQubits.size(), qubitType);
+    auto outQubitTypes = TypeRange(qubitTypes);
+
+    // DCX = CNOT(q2,q1) CNOT(q1,q2)
+    auto cnot1 = rewriter.create<catalyst::quantum::CustomOp>(
+        op.getLoc(), outQubitTypes, TypeRange{}, ValueRange{}, inQubitsValues,
+        "CNOT", false, inCtrlQubits, ValueRange{});
+
+    auto cnot2 = rewriter.create<catalyst::quantum::CustomOp>(
+        op.getLoc(), outQubitTypes, TypeRange{}, ValueRange{},
+        ValueRange{cnot1.getResult(1), cnot1.getResult(0)}, "CNOT", false,
+        inCtrlQubits, ValueRange{});
+
+    rewriter.replaceOp(op, cnot2.getResults());
+    return success();
+  }
+};
+
+template <>
+struct ConvertMQTOptSimpleGate<opt::RZXOp> final
+    : OpConversionPattern<opt::RZXOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(opt::RZXOp op, opt::RZXOp::Adaptor adaptor,
+                  ConversionPatternRewriter& rewriter) const override {
+    auto inQubitsValues = adaptor.getInQubits();
+    const auto posCtrlQubitsValues = adaptor.getPosCtrlInQubits();
+    const auto negCtrlQubitsValues = adaptor.getNegCtrlInQubits();
+    const auto theta = adaptor.getParams()[0];
+
+    SmallVector<Value> inCtrlQubits(posCtrlQubitsValues.begin(),
+                                    posCtrlQubitsValues.end());
+    inCtrlQubits.append(negCtrlQubitsValues.begin(), negCtrlQubitsValues.end());
+
+    // Output type setup
+    const Type qubitType =
+        catalyst::quantum::QubitType::get(rewriter.getContext());
+    const SmallVector<Type> qubitTypes(
+        inQubitsValues.size() + inCtrlQubits.size(), qubitType);
+    auto outQubitTypes = TypeRange(qubitTypes);
+
+    // RZX(θ) = H(q2) CNOT(q1,q2) RZ(θ)(q2) CNOT(q1,q2) H(q2)
+    auto h1 = rewriter.create<catalyst::quantum::CustomOp>(
+        op.getLoc(), outQubitTypes, TypeRange{}, ValueRange{}, inQubitsValues,
+        "H", false, inCtrlQubits, ValueRange{});
+
+    auto cnot1 = rewriter.create<catalyst::quantum::CustomOp>(
+        op.getLoc(), outQubitTypes, TypeRange{}, ValueRange{}, h1.getResults(),
+        "CNOT", false, inCtrlQubits, ValueRange{});
+
+    auto rz = rewriter.create<catalyst::quantum::CustomOp>(
+        op.getLoc(), outQubitTypes, TypeRange{}, ValueRange{theta},
+        cnot1.getResults(), "RZ", false, inCtrlQubits, ValueRange{});
+
+    auto cnot2 = rewriter.create<catalyst::quantum::CustomOp>(
+        op.getLoc(), outQubitTypes, TypeRange{}, ValueRange{}, rz.getResults(),
+        "CNOT", false, inCtrlQubits, ValueRange{});
+
+    auto h2 = rewriter.create<catalyst::quantum::CustomOp>(
+        op.getLoc(), outQubitTypes, TypeRange{}, ValueRange{},
+        cnot2.getResults(), "H", false, inCtrlQubits, ValueRange{});
```

Comment:
This (as well as some of the constructions below) cannot be correct yet as the number of qubits in the various operations is not correct. 
A Hadamard only acts on one qubit, so it cannot take both input qubits as an argument. Similarly, the CNOT can not work just on the qubit resulting from the CNOT.
Proper testing would have revealed that.

In this case, the code can even be simplified: the sequence of CNOT, Rz, CNOT forms an RZZ gate. May as well use that. And the controls only need to be applied to that RZZ gate; both H gates can remain uncontrolled because they cancel whenever the controls of the RZZ gate are not activated.